modules/face/src/face_alignment.cpp

```diff
@@ -0,0 +1,344 @@
+private:
+    int numlandmarks_;
+    int numfaces_;
+    std::vector<cv::Point2f>  meanshape_;
+};
+FacemarkKazemi::~FacemarkKazemi(){};
+bool ShapePredictor :: getBoundingRect(cv::Mat src,std::vector<cv::Rect> &facep){
+    std::vector<cv::Rect> facesp;
+    cv::CascadeClassifier face_cascade;
+    if(!face_cascade.load(face_cascade_name)){
+        CV_Error(Error::StsError, "Error while loading face detector");
+        return false;
+    }
+    Mat frame_gray;
+    cvtColor( src, frame_gray, cv::COLOR_BGR2GRAY );
+    equalizeHist( frame_gray, frame_gray );
+    face_cascade.detectMultiScale( frame_gray, facep, 1.05, 3, 0, Size(30,30) );
+    return true;
+}
+
+
+bool ShapePredictor :: getData(std::vector<string> filename,std::vector<std::vector<cv::Point2f> >
+                              &_trainlandmarks,std::vector<string> &_trainimages)
+{
+    for(unsigned long j=0;j<filename.size();j++){
+        ifstream f1(filename[j].c_str());
+        if(!f1.is_open()){
+            CV_Error(Error::StsError, "File can't be opened for reading!");
+            return false;
+        }
+        string img;
+        //get the path of the image whose landmarks have to be detected
+        getline(f1,img);
+        //cout<<img;
+        //push the image paths in the vector
+       _trainimages.push_back(img);
+        std::vector<cv::Point2f> temp;
+        string s;
+        while(getline(f1,s)){
+            stringstream ss1,ss2;
+            unsigned long i=0;
+            char x1[100],y1[100];
+            while(s[i]!=','){
+                x1[i]=s[i];
+                i++;
+            }
+            string x,y;
+            x=string(x1);
+            Point2f pt;
+            ss1<<x;
+            ss1>>pt.x;
+            ss1.clear();
+            i++;
+            int p=0;
+            while(i<s.length()){
+                y1[p++]=s[i];
+                i++;
+            }
+            y=string(y1);
+            ss2<<y;
+            ss2>>pt.y;
+            ss2.clear();
+            //cout<<pt.x<<" "<<pt.y<<endl;
+           temp.push_back(pt);
+        }
+        _trainlandmarks.push_back(temp);
+        f1.close();
+    }
+    return true;
+}
+bool ShapePredictor :: getMeanShapeRelative(std::vector<cv::Rect> face,std::vector<vector<Point2f> > & initialshape_){
+    if(meanshape_.empty()) {
+            // throw error if no data (or simply return -1?)
+            String error_message = "The data is not loaded properly by train function. Aborting...";
+            CV_Error(Error::StsBadArg, error_message);
+            return false;
+    }
+    //calculatinng source points for warping
+    Point2f srcTri[3];
+    float minx=8000.0,maxx=0.0,miny=8000.0,maxy=0.0;
+    for(unsigned long i=0;i<meanshape_.size();i++){
+        Point2f pt1;
+        pt1.x=meanshape_[i].x;
+        if(pt1.x<minx)
+            minx=pt1.x;
+        if(pt1.x>maxx)
+            maxx=pt1.x;
+        pt1.y=meanshape_[i].y;
+        cout<<pt1.x<<","<<pt1.y<<endl;
```

Comment:
A function should not through a lot of stuff to the cout. This is debugging code and should be removed in your PR.